### PR TITLE
Modify Notifications

### PR DESCRIPTION
**Pull request by VetalKamen**
Added To Notifications :viewed, :after_remove and render a show_notifications layout in the dropdown

### CODE REVIEW

File: app/controllers/notifications_controller.rb [hunk 1]
```diff
@@ -1,22 +1,43 @@
...
+  before_action :logged_in_user, only: [:destroy, :toggle_action, :toggle_menu_action]
+  before_action :correct_user, only: [:destroy, :toggle_action]
 
   def destroy
     @notification.destroy
     flash[:success] = 'Notification deleted'
     redirect_to request.referrer || root_url
   end
 
+  def toggle_action
```
Comment (Kasvit) [inline app/controllers/notifications_controller.rb:11]: why `toggle_action` ?)
maybe better to name like `toggle_viewed` ?)

File: app/controllers/notifications_controller.rb [hunk 2]
```diff
@@ -1,22 +1,43 @@
...
+      end
+    else
+      respond_to do |format|
+        format.html { redirect_to request.referrer }
+      end
+    end
+  end
+
+  def toggle_menu_action
+    @notifications = current_user.notifications.order(viewed: :desc, updated_at: :desc)
```
Comment (Kasvit) [inline app/controllers/notifications_controller.rb:26]: can it be the same as `:index` action?)

File: app/models/notification.rb [hunk 3]
```diff
@@ -1,5 +1,7 @@
 class Notification < ApplicationRecord
   belongs_to :user
   validates :user_id, presence: true
+  validates :target_id, presence: true
   validates :text, presence: true
+  scope :unread, -> { where('viewed=>false') }
```
Comment (Kasvit) [inline app/models/notification.rb:6]: try with `where(viewed: false)`

File: app/views/layouts/_header.html.slim [hunk 4]
```diff
@@ -24,6 +24,10 @@ body
                 li.divider
                 li
                   = link_to "Log out", logout_path, method: :delete
+            li.dropdown.notifications
+              = link_to 'Notifications', toggle_menu_action_path(current_user), method: :post, id: 'notifications-menu', class: 'dropdown-toggle', :data => { :toggle => 'dropdown' }, remote: true
+              ul.dropdown-menu.notifications-menu
```
Comment (Kasvit) [inline app/views/layouts/_header.html.slim:29]: please add some screenshots?

File: app/controllers/notifications_controller.rb [hunk 5]
```diff
@@ -1,22 +1,40 @@
...
 
   def destroy
     @notification.destroy
     flash[:success] = 'Notification deleted'
     redirect_to request.referrer || root_url
   end
 
+  def toggle_viewed
+    @user = @notification.user
+    if @notification.update_attribute('viewed', '1')
```
Comment (Kasvit) [inline app/controllers/notifications_controller.rb:13]: try `@notification.update(viewed: '1')`

File: app/views/layouts/_header.html.slim (hunk 4 above)
Comment (VetalKamen) [reply app/views/layouts/_header.html.slim:29]: image
image

**Pull request merged**